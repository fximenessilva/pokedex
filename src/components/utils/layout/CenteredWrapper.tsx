import React from 'react';
import styled from 'styled-components';

import { ComponentWithDarkMode } from '../../../utils/types';

const CenteredWrapper = styled.div<ComponentWithDarkMode>`
  display: flex;
  justify-content: center;
  align-items: center;
  width: calc(100% - 4rem);
  padding: 2rem;
`;

export default CenteredWrapper;
