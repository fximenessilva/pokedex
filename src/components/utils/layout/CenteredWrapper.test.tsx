import React from 'react';
import { render } from '@testing-library/react';
import CenteredWrapper from './CenteredWrapper';

describe('CenteredWrapper', () => {
  it('renders correctly with dark mode', () => {
    const { container } = render(<CenteredWrapper $isDarkMode={true} />);

    // Assert that the rendered component matches the snapshot
    expect(container.firstChild).toMatchSnapshot();

    // You can add more assertions as needed to check styles or elements
    // For example, you can check the background color:
    // expect(container.firstChild).toHaveStyle('background-color: #424B54');
  });

  it('renders correctly with light mode', () => {
    const { container } = render(<CenteredWrapper $isDarkMode={false} />);

    // Assert that the rendered component matches the snapshot
    expect(container.firstChild).toMatchSnapshot();

    // You can add more assertions as needed to check styles or elements
    // For example, you can check the background color:
    // expect(container.firstChild).toHaveStyle('background-color: #E5ECF3');
  });
});
