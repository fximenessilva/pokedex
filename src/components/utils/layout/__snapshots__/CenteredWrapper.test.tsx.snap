// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CenteredWrapper renders correctly with dark mode 1`] = `
<div
  class="sc-beyTiQ gdHqrM"
/>
`;

exports[`CenteredWrapper renders correctly with light mode 1`] = `
<div
  class="sc-beyTiQ bLjOgq"
/>
`;
